### DISCUSSION TITLE
How use nested routes in Atomic Design

### DISCUSSION THREAD

**Discussion by JSsantana** [remix-run/remix#2847] [Q&A]
How can I use nested routes when I have a main template that gets its children and props components? Example:

```
import React from 'react'

import {
  PageTemplate, Header, Hero, Footer, FeatureList,
} from 'components'

const HomePage = () => {
  return (
    <PageTemplate
      header={<Header />} // should be nested route
      hero={<Hero />} // should be nested route
      footer={<Footer />} // should be nested route
    >
      <FeatureList />
    </PageTemplate>
  )
}

export default HomePage
```

**Comment by JSsantana** [comment]
Sorry for the title. It shoud be about routing child components as props

**Comment by sergiodxa** [comment]
Think of URLs, you have `/parent/child` right? So you can only have one child at the same, you could of course have more child routes inside the other child route like `/grandparent/parent/child`.

This means, a single route can't have two nested routes at the same time, grandparent only has parent as nested and parent only has child.

There's no way in the browser to have something like
```
/parent
  /child1
  /child2
```
And render both children routes at the same time.

With that said, if you want to make Header, Hero and Footer be different nested routes you would have to create multiple levels of nested routes like `/home/header/hero/footer`, you could use pathless layout routes to avoid adding segments to the URL too.

Another option, is to use the `handle` export and `useMatches` hook to let the nested route export the main content as the default export and from the handle pass the header, hero and footer.
